core/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -150,13 +150,9 @@ private String innerDetect() throws Exception {
         ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
         boolean enabledCpu = false;
         try {
-            if (threadBean.isThreadCpuTimeSupported()) {
-                if (!threadBean.isThreadCpuTimeEnabled()) {
-                    enabledCpu = true;
-                    threadBean.setThreadCpuTimeEnabled(true);
-                }
-            } else {
-                throw new IllegalStateException("MBean doesn't support thread CPU Time");
+            if (!threadBean.isThreadCpuTimeEnabled()) {
+                enabledCpu = true;
+                threadBean.setThreadCpuTimeEnabled(true);
```

Comment:
Somewhat related to this PR: should we just remove this code? this "save/restore" is quite complicated, and I know for a fact it never happens (we don't have any OS we have ever tested on where thread cpu time is *supported* but not *enabled*, and if such a situation were to present itself, we'd surely want to know what the implications of switching it on temporarily are).

I know this because this method requires permissions that we don't grant `ManagementPermission("control")`, so it does not ever succeed.

We don't have to do this on this PR though, but it definitely makes the code here complex.